### DISCUSSION TITLE
My two cents regarding SSO integration

### DISCUSSION THREAD

**Discussion by danielr1996** [dani-garcia/vaultwarden#4732] [General]
Hey folks, if you're interested in using Vaultwarden with SSO you've probably stumbled across the many discontinued pull requests and the still open pull requests #3899 which has 310 comments as of today. Maybe you read through all those, maybe you've given up. I spent the last couple hours digging through all of them and to spare you from that I'm going to summarize it for you, maybe we can get this pinned so it's better visible or create a dedicated page in the wiki.

# What happened and why is it taking so long?
A few years ago bitwarden added OAuth support to the official bitwarden server, soon a few PR's/Forks popped up and added rudimental support but were far from mergeable and then discontinued, the maintainers got pretty confused about which PR' to review and didn't have enough time to handle multiple concurring PR's. Fast forward to now:

 @Timshel maintains a fork at https://github.com/Timshel/vaultwarden. As far as I can tell and read from the comments the fork is still regurarely synced with the upstream but @Timshel doesn't plan to maintain this fork forever and want's to get it merged _eventually_, but for now it works as a drop in replacement.  The maintainers of the repository (for this PR mainly @BlackDex) are willing to merge this PR _eventually_ but still have concerns regarding backwards compatibility.

# Why is this taking so long?
While SSO Integration sure would be a nice benefit for those already using oauth it's probably not worth for the majority of users. The benefit of using oauth comes with the cost of e.g. hosting a separate keycloak instance and learning about oauth.
If you are selfhosting for your family and friends you might have up to 20 users, which in my opinion isn't justified at that scale and there might not be high demand for this across the whole userbase.

On the otherhand this is a complex topic, it's not about fixing a display bug or changing the theme. It's about the central authentication mechanism where bugs can have a high impact on the security and availability of the application. 
 
# What can I do to help getting this feature into vaultwarden?
## Be respectfull
Please beware that this is a community driven open source project, no one here owes you _anything_ , no matter how important something might be for you personally. So please be respectfull with those investing their time in this project.

Please stop commenting "please merge", "I need this because x", etc. in the issues, this makes following the events really hard for contributors and further delays the merge. 

## Contribute
This is an open source project, feel free to contribute. Most of the work is already done but we're not quite there. See the [PR's discussion](https://github.com/dani-garcia/vaultwarden/pull/3899) and the forks [issues](https://github.com/Timshel/vaultwarden/issues) to see what still needs to be done.

## Donate
Go check out the contributors profiles If they have a sponsors button and leave a tip to thank them for keeping this topic alive for so long. 

# Disclaimer
This is not a official statement from the maintainers but my personal summary, please let me know If I got something wrong.

**Comment by Timshel** [comment]
Hey,

First thanks for taking the time to read all the PR comment, making this and the sponsor :smile:.

One minor correction: I believe the main blocker is that the maintainers don't have the time at the moment to review such a big change (keeping track with Bitwarden evolution already take quite a bit of time). 

And not to start a debate but even with 20 users SSO might be useful if you are hosting multiple applications ;) (that's my use case ^^); but as you mention it will to not be useful to the majority of users.

Cheers.
